code/routers/user/reset.js

```diff
@@ -0,0 +1,203 @@
+const express = require('express');
+const passport = require('passport');
+const router = express.Router();
+
+
+const User = require('../../models/user');
+const resCode = require('../../config/resCode');
+
+var async = require("async");
+var nodemailer = require("nodemailer");
+var crypto = require("crypto");
+
+/*
+  forgot password api
+
+  let's users reset their password and redirect
+  them to the login page
+*/
+
+router.get('/forgot', function(req, res) {
+  res.redirect('forgot');
+});
+
+router.post('/forgot', function(req, res, next) {
+  console.log("indise reset/forgot router")
+  async.waterfall([
+    function(done) {
+      crypto.randomBytes(20, function(err, buf) {
+        var token = buf.toString('hex');
+        done(err, token);
+      });
+    },
+    function(token, done) {
+      User.findOne({ email: req.body.email }, function(err, user) {
+        if (!user) {
+          //req.flash('error', 'No account with that email address exists.');
+          //return res.redirect('/forgot');
+        }
+
+        user.resetPasswordToken = token;
+        user.resetPasswordExpires = Date.now() + 3600000; // 1 hour
+
+        user.save(function(err) {
+          done(err, token, user);
+        });
+      });
+    },
+    function(token, user, done) {
+      var smtpTransport = nodemailer.createTransport({
+        host: "smtp.gmail.com",
+        auth: {
+          type: "login",
+          user: "DOMAD24901@gmail.com",
+          pass: "DoMADTemp#2020",
+        }
+      });
+
+      var mailOptions = {
+        to: user.email,
+        from: 'DOMAD24901@gmail.com',
+        subject: 'Node.js Password Reset',
+        text: 'You are receiving this because you (or someone else) have requested the reset of the password for your account.\n\n' +
+          'Please click on the following link, or paste this into your browser to complete the process:\n\n' +
+          'http://localhost:' + 3000 + '/' + token + '\n\n' +
+          'If you did not request this, please ignore this email and your password will remain unchanged.\n'
+      };
+      smtpTransport.sendMail(mailOptions, function(err) {
+        console.log('mail sent');
+        console.log('email: ' + user.email)
+        //req.flash('success', 'An e-mail has been sent to ' + user.email + ' with further instructions.');
+        done(err, 'done');
+      });
+    }
+  ], function(err) {
+    if (err) return next(err);
+    res.redirect('/forgot');
+  });
+});
+
+// :token -> check-token
```

Comment:
Do we still want to change this api call to `check-token` instead of `:token` to make it a little more clear?

Reply:
I think this can be done like we've discussed before, but I was having some problems on the react side of things because the token can only be retrieved from the url. I'll give it try and get back to you.

Reply:
The <code>/forgot</code> api appends the token like this:
<code>'http://localhost:' + 3000 + '/' + token</code>
The token variable is being passed to <code>/:token</code> and there's really no way around this. I've trying changing things by passing it trough a hidden input variable from the front-end side, but the page doesn't render. 
Plus the token variable expires after 1 hour, so I don't think we need to worry about having it on the url.